From 8431cabd4d8f72dfaadbe64adf9f8dc89a26ae62 Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Thu, 6 Sep 2012 17:40:51 +0200
Subject: [PATCH] EphyOverviewStore: don't register the snapshot mtime as long

All code paths use int, and this breaks on architechtures where
long is not the same as int (such as x86_64).

https://bugzilla.gnome.org/show_bug.cgi?id=683029

[Do the same for the visit time column]
Signed-off-by: Claudio Saavedra <csaavedra@igalia.com>
---
 lib/widgets/ephy-overview-store.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/widgets/ephy-overview-store.c b/lib/widgets/ephy-overview-store.c
index 99ef2cf..074ec8b 100644
--- a/lib/widgets/ephy-overview-store.c
+++ b/lib/widgets/ephy-overview-store.c
@@ -161,10 +161,10 @@ ephy_overview_store_init (EphyOverviewStore *self)
   types[EPHY_OVERVIEW_STORE_TITLE] = G_TYPE_STRING;
   types[EPHY_OVERVIEW_STORE_AUTHOR] = G_TYPE_STRING;
   types[EPHY_OVERVIEW_STORE_SNAPSHOT] = GDK_TYPE_PIXBUF;
-  types[EPHY_OVERVIEW_STORE_LAST_VISIT] = G_TYPE_LONG;
+  types[EPHY_OVERVIEW_STORE_LAST_VISIT] = G_TYPE_INT;
   types[EPHY_OVERVIEW_STORE_SELECTED] = G_TYPE_BOOLEAN;
   types[EPHY_OVERVIEW_STORE_SNAPSHOT_CANCELLABLE] = G_TYPE_CANCELLABLE;
-  types[EPHY_OVERVIEW_STORE_SNAPSHOT_MTIME] = G_TYPE_LONG;
+  types[EPHY_OVERVIEW_STORE_SNAPSHOT_MTIME] = G_TYPE_INT;
   types[EPHY_OVERVIEW_STORE_CLOSE_BUTTON_RENDER_POLICY] = EPHY_TYPE_REMOVABLE_PIXBUF_RENDER_POLICY;
 
   gtk_list_store_set_column_types (GTK_LIST_STORE (self),
-- 
1.7.12

